product_restricted_type/models/product_template.py

```diff
@@ -1,27 +1,29 @@
-# Copyright 2018 ForgeFlow S.L.
+# Copyright 2018-2024 ForgeFlow S.L.
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
-from odoo import _, api, models
+from odoo import _, api, fields, models
 from odoo.exceptions import ValidationError
 
 
 class ProductTemplate(models.Model):
     _inherit = "product.template"
 
+    allowed_categ_ids = fields.Many2many(
+        "product.category",
+        compute="_compute_allowed_categ_ids",
+        string="Allowed Categories",
+    )
+
     @api.onchange("categ_id")
     def _onchange_categ_id(self):
```

Comment:
You should use _compute_type instead: https://github.com/odoo/odoo/blob/17.0/addons/product/models/product_template.py#L58

Reply:
thanks for the suggestion, let me know if that's what you meant to do. 👍🏿

Reply:
After double-checking I have added back the onchnage method back again, I added a comment in order to explain what's the purpose of the method, let me know if you agree 🙂